frontend/src/components/Follow/Follow.jsx

```diff
@@ -0,0 +1,50 @@
+import "./Follow.css";
+import axios from "axios";
+
+export default function Follow({ ticker, stockData, setStockData }) {
+    async function handleFollow() {
+        try {
+            const response = await axios.post(
+                "http://localhost:3000/follow",
+                { ticker },
+                { withCredentials: true, validateStatus: () => true }
+            );
+
+            if (response.status === 200) {
+                setStockData({ ...stockData, following: true });
+            }
+
+            if (response.status === 409) {
+                alert(response.data.error);
+            }
+        } catch (error) {}
+    }
+
+    async function handleUnfollow() {
+        try {
+            const response = await axios.post(
+                "http://localhost:3000/unfollow",
+                { ticker },
+                { withCredentials: true, validateStatus: () => true }
+            );
+
+            if (response.status === 200) {
+                setStockData({ ...stockData, following: false });
+            }
+
+            if (response.status === 404) {
+                alert(response.data.error);
+            }
+        } catch (error) {}
+    }
+
+    return (
+        <div className="follow">
+            {stockData.following ? (
+                <button onClick={handleUnfollow}>Following</button>
+            ) : (
+                <button onClick={handleFollow}>Follow</button>
+            )}
+        </div>
+    );
+}
```

Comment:
Here's an alternative way to implement follow/unfollow button -- 

Note that the `handleFollow` and `handleUnfollow` are largely duplicated logic, only real difference is a `following` boolean that you pass into the stock data and the endpoint. Best practice is we like to avoid duplicate code, so we can consolidate these 2 functions.

Instead of returning 2 different button components based on `stockData.following`, use a conditional inside of one button component to change the text, something like
```
<button onClick={handleFollow}>{isFollowing ? "Following" : "Follow"}</button>
```

You can define `isFollowing` as state using useState, and initializing it with value `stockData.following` 

Then create a handler invoked on button click to update the state.